Pronouns & custom pronouns

DeadGirlRisen opened the pull request:
Bingle (dinofelisdruid) and I have worked on adding in the pronouns and custom pronouns. One of the people said we should submit it.

What this does:
initializes cats with pronouns, mostly using the preexisting code.
Makes sure genders are saved and can be loaded up.
Changes the specify gender button, it will now allow you to manually input custom pronouns.

With this change, cats can now have pronouns of a users choice and gender alignment, and the game will use these genders with the proper gender tagging in place.

DeadGirlRisen committed:
Update lineart.png

1/21 Sprites done

DeadGirlRisen committed:
Testing Color Alignment.

DeadGirlRisen committed:
Hunter Thoughts

Empty Hunter Thoughts

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
Floof Sprites Done

DeadGirlRisen committed:
Merge branch 'development' of https://github.com/DeadGirlRisen/Lunar-Mod into development

DeadGirlRisen committed:
Infinite Rerolls

DeadGirlRisen committed:
Lots of little changes

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
Names, Colours, patrols

DeadGirlRisen committed:
Merge branch 'development' of https://github.com/DeadGirlRisen/Lunar-Mod into development

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
stuff

DeadGirlRisen committed:
Updates

DeadGirlRisen committed:
Merge branch 'development' of https://github.com/DeadGirlRisen/Lunar-Mod into development

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
pronouns

DeadGirlRisen committed:
Merge branch 'development' of https://github.com/DeadGirlRisen/Lunar-Mod into development

DeadGirlRisen committed:
pronouns

DeadGirlRisen committed:
Pronoun Update

I'm not very good with github, so um, hopefully this doesn't actually say its changing 615 files, because its like, 4 files.

DeadGirlRisen committed:
Custom Pronouns part 2

Makes sure that if cat is female/male/trans male/trans female that they can still have custom pronouns.

DeadGirlRisen committed:
Pronoun

keyraven commented on scripts/cat/cats.py:
```diff
@@ -139,7 +139,7 @@ def __init__(self,
             self.adoptive_parents = []
             self.mate = []
             self.status = status
-            self.pronouns = [self.default_pronouns[0].copy()]
```
This should be left. Every cat need to have pronouns, and in case there is an error with the assignment later, a cat should at least have one default pronoun.

DeadGirlRisen replied:
Alright, I got this part added in, so I believe this comment can be resolved now. Working on the other stuff still with Bingle.

keyraven commented on scripts/cat/cats.py:
```diff
@@ -280,32 +280,41 @@ def __init__(self,
                     self.genderalign = "trans female"
+                    self.pronouns = [self.default_pronouns[1].copy()]
                 elif nb_chance == 1:
                     self.genderalign = "nonbinary"
```
There is no pronoun assignment for nonbinary cats.

DeadGirlRisen replied:
Alright, I got this part added in, so I believe this comment can be resolved now. Working on the other stuff still with Bingle.

keyraven commented on scripts/game_structure/load_cat.py:
```diff
@@ -121,6 +121,7 @@ def json_load():
             new_cat.adoptive_parents = cat["adoptive_parents"] if "adoptive_parents" in cat else []
             
             new_cat.genderalign = cat["gender_align"]
+            new_cat.pronouns = cat["pronouns"]
```
This causes a load-error when for older saves.  To fix this issue, use `.get` , with an appropriate default.

DeadGirlRisen replied:
got this done! One of the easier things.

keyraven commented on scripts/game_structure/windows.py:
```diff
@@ -462,14 +462,14 @@ class SpecifyCatGender(UIWindow):
 
     def __init__(self, cat):
-        super().__init__(scale(pygame.Rect((600, 430), (800, 370))),
+        super().__init__(scale(pygame.Rect((560, 360), (845, 470))),
```
Regarding UI - I think this UI is too simple, and not very user friendly. There are no labels on the pronoun entry boxes, which may be confusing. In addition, there is no way to select from default pronouns, nor a way to add multiple pronouns.

DeadGirlRisen replied:
We got it so you can select default pronouns, custom pronouns, have multiple pronouns. There's labels on the pronoun entry boxes. And an example box so you can test pronouns.

And we have a setting where it starts with all cats being they/them and you can turn it off.

Our customized pronouns is also an option to turn on or off, in case you just want the quick gender alignment.

keyraven reviewed (changes requested):
In addition to my other comments, I am wary of pushing pronouns without a setting to refer to all cats as "they/them". That setting was promised, and we will have some upset players if the feature is pushed without the setting.

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
Show Examples

Button

DeadGirlRisen committed:
Merge branch 'Pronouns-&-Custom-Pronouns' of https://github.com/DeadGirlRisen/Lunar-Mod into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
Small changes

The small changes requested, Including default pronouns as they/them as a fallback, and making sure nonbinary has the setting pronouns.

Also corrected a mistake where I didn't call upon the right conju.

DeadGirlRisen committed:
little changes pt2

Realized I messed up the nonbinary codings. Corrected now.

DeadGirlRisen committed:
Update relationship.py

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
Optional Custom Pronouns (WIP)

Two settings to activate using gendered pronouns, currently non functional as its being worked on.

the other is the ability to toggle custom pronouns for the specify gender button.

DeadGirlRisen committed:
They/them default

Cats start with they/them as a default setting

DeadGirlRisen committed:
Wording fix

DeadGirlRisen committed:
Wording pt2

DeadGirlRisen committed:
Old Saves

 are able to get they/them pronouns now.

DeadGirlRisen committed:
Setting up for multiple pronouns

Adding an identifier here, just in case.

DeadGirlRisen committed:
Pelts

DeadGirlRisen committed:
Pronoun Work

Still WIP.
Allows multiple pronouns to be set to a cat.
Allows a user to test the pronouns.
Shows a list of pronouns the cat has.

Major thanks to bingle.

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into development

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
Update rosettecolours.png

DeadGirlRisen committed:
Fixing text cutoff

DeadGirlRisen committed:
Update windows.py

DeadGirlRisen committed:
moving buttons

such fun

DeadGirlRisen committed:
Images

A lot of the images required for the screen.

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
buttons

Why are the hover affects not working? :thinking:

DeadGirlRisen committed:
Merge branch 'Pronouns-&-Custom-Pronouns' of https://github.com/DeadGirlRisen/Lunar-Mod into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
Update ChangeGenderScreen.py

DeadGirlRisen committed:
Removing Pronouns

Allows user to remove pronouns.

DeadGirlRisen committed:
Fixing a window issue

DeadGirlRisen committed:
loading presets

DeadGirlRisen committed:
Merge branch 'ClanGenOfficial:development' into Pronouns-&-Custom-Pronouns

DeadGirlRisen committed:
Bug Fixing

Working on a few minor bugs and cosmetic issues, but were in the final stretch.

If your reading this as im working on it. idk. you might not check on it until I pull it out of draft mode and sent in a comment saying im ready for our next inspection XD

DeadGirlRisen committed:
Bug fixes

And this should be it, next is just cleaning up some print debuging statments.

DeadGirlRisen committed:
Cleanup

Cleaning up some print statments.

DeadGirlRisen commented:
Alright! Changes have been addressed! Bingle and I have worked hard to get it all!

DeadGirlRisen committed:
Merge branch 'development' into Pronouns-&-Custom-Pronouns

keyraven commented on resources/dicts/events/ceremonies/ceremony_traits.json:
```diff
@@ -593,5 +593,23 @@
         "judgement",
         "foresight"
+    ],
+    "cruel": [
```
Did you add a trait? If you wish to propose an additional trait, please remove and PR this separately

keyraven commented on resources/dicts/names/names.json:
```diff
@@ -14,29 +14,26 @@
-    "thistle", "thorn", "thrift", "throat", "thrush", "thud", "thunder", "tiger", "timber", "toe", "tooth", "trail", "tree", 
-    "trot", "tumble", "twist", "valley", "watcher", "water", "wave", "web", "whisper", "whistle", "willow", "wind", "wing", "wish", "wisp", 
-    "zoom"
+    "arch", "ash", "aster", "back", "badger", "bark", "bat", "beak", "beam", "bee", "beetle", "bellow", "belly", "berry", "billow", "bird", "bite", "blaze", "blink", "bloom",
```
Why were changes made to the names? If you wish to propose additional names, please PR separately.

keyraven commented on resources/dicts/patrols/general/medcat.json:
```diff
@@ -50,428 +65,1376 @@
-                "text": "s_c makes sure everyone gets a good basking spot as the cats pause for a little break before resuming their herb-gathering.",
-                "exp": 20,
-                "weight": 20,
+                "text": "The sunlight feels great and s_c leads the cats in conversation, bringing them closer together.",
```
Please revert any added patrols. These need to be put in a separate PR.

keyraven commented on resources/dicts/pronouns.json:
```diff
@@ -0,0 +1,42 @@
+{
+    "default_pronouns": [
+        {
+            "name": "Neutral",
```
Can you explain the purpose of the pronoun "name"?

keyraven commented on scripts/cat/cats.py:
```diff
@@ -75,8 +75,10 @@ class Cat():
+
     default_pronouns = [
         {
+            "name": "Neutral",
```
Are default pronouns defined here, or in the JSON resource file? These should only be defined once.

keyraven commented on scripts/events_module/relationship/pregnancy_events.py:
```diff
@@ -765,8 +765,15 @@ def get_amount_of_kits(cat):
+        eight_kits = [min_kits + 7] * game.config["pregnancy"]["eight_kit_possibility"][cat.age]
+        nine_kits = [min_kits + 8] * game.config["pregnancy"]["nine_kit_possibility"][cat.age]
+        ten_kits = [min_kits + 9] * game.config["pregnancy"]["ten_kit_possibility"][cat.age]
+        eleven_kits = [min_kits + 10] * game.config["pregnancy"]["eleven_kit_possibility"][cat.age]
```
We do not wish to increase the possible number of kits to 11. Please revert.

keyraven reviewed (changes requested):
I am unable to access the pronoun change screen, so I can't give a proper review. 

Notes: 

- All the sprites have been replaced with a high-res texture pack, which currently render as black silhouettes.  Please restore the official spritesheets. 
- You also add a lot of thoughts, patrols, patrol art, skill paths. and an additional trait. Please revert these changes and PR them separately if you wish to propose these changes. 
- Unit tests and pronoun test are failing.

DeadGirlRisen commented:
Ack! I'm so sorry! I don't know what happend?

It must have somehow pulled from my Lunar Mod?

I am so sorry!

Github is a bit confusing for me, so I must have messed up

DeadGirlRisen reviewed (commented):
How do I undo this thing here?

DeadGirlRisen committed:
Delete resources/dicts/events/ceremonies/ceremony_traits.json

DeadGirlRisen commented:
Okay, So I don't know how to undo the accidental merge, but like, I know what files I messed with generally, so I'm going to create a clean branch again, put the files back, then it should be good to go. Im so sorry, I was trying to add this onto my lunar mod, but I must have somehow did this both ways?

DeadGirlRisen commented:
Okay, I'm closing this, because I messed up, however I will reopen with a clean version, because im not sure how to fix this, im so sorry Keyraven for all the confusion and mess.
